test/new-e2e/tests/agent-platform/package-signing/package_signing_test.go

```diff
@@ -0,0 +1,68 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+package packagesigningtest
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/client"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/params"
+	"github.com/stretchr/testify/assert"
+
+	"testing"
+)
+
+type Payload struct {
```

Comment:
Does it have to be public?

Reply:
Maybe not (and it not all the metadata payloads should be changed, as I based my implementation on `inventoryhost.go`)

Reply:
We could change the visibility of the other structs on a separate PR 😄

Reply:
Actually it needs to be public if I want to benefit from it to unmarshall in the e2e tests